dotnet/targets/Xamarin.Shared.Sdk.targets

```
@@ -381,9 +381,15 @@
 	This doesn't work after linking with these libraries, so this must be fixed to be @executable_path/. Eventually mono will ship dylibs we don't have to fix (https://github.com/dotnet/runtime/issues/34637).
 	-->
 	<Target Name="_UpdateDynamicLibraryId" DependsOnTargets="_ComputeVariables" Inputs="@(_MonoLibrary)" Outputs="@(_MonoLibrary -> '$(_IntermediateNativeLibraryDir)%(Filename)%(Extension)')">
+		<PropertyGroup>
+			<_ExecutablePathPrefix Condition="'$(_PlatformName)' != 'macOS'">@executable_path/</_ExecutablePathPrefix>
+			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'macOS'">@executable_path/../MonoBundle/</_ExecutablePathPrefix>
```

Developers can use their own name instead of `MonoBundle`, so this is necessary:

```suggestion
			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'macOS'">@executable_path/../$(_CustomBundleName)/</_ExecutablePathPrefix>
```

---

Happy to change that. There was one other place in the file where it was hard-coded which is where I copied it from.

---

Apparently `_CustomBundleName` is not set at this point. I'll look into it tomorrow.

---

> Apparently `_CustomBundleName` is not set at this point. I'll look into it tomorrow.

You can add a dependency on the `_ParseBundlerArguments` target to fix that.

> There was one other place in the file where it was hard-coded which is where I copied it from.

Yes, I still have to update a few existing places.

---

`_ParseBundlerArguments` doesn't exist in the dotnet builds. There's only `_ComputeLinkerArguments` which doesn't parse `CustomBundleName` yet.

---

Thanks for the `_ParseBundlerArguments` tip. Apparently my VS Code search is broken and it doesn't recurse in some directories. :-/